src/NuGet.Core/NuGet.Packaging/Signing/TrustStore/FallbackCertificateBundleX509ChainFactory.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if NET5_0_OR_GREATER
+
+using System;
+using System.IO;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+
+namespace NuGet.Packaging.Signing
+{
+    internal sealed class FallbackCertificateBundleX509ChainFactory : CertificateBundleX509ChainFactory
+    {
+        internal const string FileName = "cstsroots.pem";
+
+        private static readonly Lazy<string> ThisAssemblyDirectoryPath = new(GetThisAssemblyDirectoryPath, LazyThreadSafetyMode.ExecutionAndPublication);
+
+        private FallbackCertificateBundleX509ChainFactory(X509Certificate2Collection certificates, string filePath)
+            : base(certificates, filePath)
+        {
+        }
+
+        internal static bool TryCreate(out FallbackCertificateBundleX509ChainFactory factory, string filePath = FileName)
+        {
+            factory = null;
+
+            string fullFilePath = Path.IsPathRooted(filePath) ? filePath : Path.Combine(ThisAssemblyDirectoryPath.Value, filePath);
```

Comment:
`Path.Combine()` will already do the right thing if any of these paths are absolute (rooted) so you probably don't need to check first.

https://docs.microsoft.com/en-us/dotnet/api/system.io.path.combine
> If path2 does not include a root (for example, if path2 does not start with a separator character or a drive specification), the result is a concatenation of the two paths, with an intervening separator character. If path2 includes a root, path2 is returned.